packages/firestore/src/local/local_store.ts

```diff
@@ -1400,10 +1421,22 @@ export async function saveNamedQuery(
           transaction,
           newTargetData
         );
+        localStoreImpl.targetDataByTarget = localStoreImpl.targetDataByTarget.insert(
+          newTargetData.targetId,
+          newTargetData
+        );
       }
-      return updateReadTime.next(() =>
-        localStoreImpl.bundleCache.saveNamedQuery(transaction, query)
-      );
+      return updateReadTime
+        .next(() =>
+          localStoreImpl.bundleCache.saveNamedQuery(transaction, query)
+        )
+        .next(() =>
+          localStoreImpl.targetCache.addMatchingKeys(
```

Comment:
I think you should only do this if you updated the readTime above. On top of that, you need to also remove all existing keys, otherwise your mapping will not match the backend's mapping.

Reply:
Agree to both.